client/src/driver/BitBakeProjectScanner.ts

```diff
@@ -390,23 +390,29 @@ You should adjust your docker volumes to use the same URIs as those present on y
     // Some recipes change their PN like gcc-source -> gcc-source-13.2
     // We allow the recipe to be found by just including part of the name
     const recipesWithoutPaths = this._bitbakeScanResult._recipes.filter((recipe) => recipe.path === undefined)
+    // Sort by increasing length to avoid partial matches
+    matches.sort((a, b) => a.length - b.length)
+    recipesWithoutPaths.sort((a, b) => a.name.length - b.name.length)
     await this.assignRecipesPaths(matches, recipesWithoutPaths, (a: string, b: string) => a.includes(b))
   }
 
   private async assignRecipesPaths (filePaths: string[], recipesArray: ElementInfo[], nameMatchFunc: (a: string, b: string) => boolean): Promise<void> {
-    for (const filePath of filePaths) {
+    for (let i = 0; i < filePaths.length; i++) {
+      const filePath = filePaths[i]
       const recipePath = await this.resolveContainerPath(filePath.trim()) as string
       const recipeName = extractRecipeName(recipePath)
       const recipeVersion = extractRecipeVersion(recipePath)
       const recipe = recipesArray.find((element: ElementInfo): boolean => {
-        return nameMatchFunc(element.name, recipeName)
+        return element.path === undefined && nameMatchFunc(element.name, recipeName)
```

Comment:
This is actually needed otherwise the go-cross recipes don't all have their .bb because previous ones in the loop match. It's sort of a "recipesWithoutPaths" mechanism within the same assignRecipesPaths pass.

Reply:
I see, because it is traversing the filePath. If it is looping against the recipes without paths, we don't have to put the redundant expression to filter out recipes that have been assigned a path in the previous loops. Also, it reduces the number of loops.

Reply:
@WilsonZiweiWang Are you ok with the current PR? You may approve if OK.